docs/machine-learning/Convolutional_Neural_Network.md

```diff
@@ -0,0 +1,176 @@
+
+4. **Transposed Convolution**:  
+   Used in tasks like image generation, transposed convolutions perform the opposite operation of standard convolutions, increasing the spatial dimensions of the input.
+
+### Problem Statement:
+Given an image dataset, the goal of a CNN is to classify the images into different categories (e.g., classifying digits in the MNIST dataset or identifying objects in CIFAR-10). CNNs are also used in segmentation, detection, and generation tasks in computer vision.
+
+### Key Concepts:
+- **Filters (Kernels)**:  
+  Filters are small matrices that slide over the input data to detect features like edges, corners, or textures. Multiple filters are used to detect various features.
+
+- **Stride**:  
+  The number of pixels by which the filter slides over the input data. A larger stride reduces the spatial dimensions of the feature map.
+
+- **Padding**:  
+  Adding pixels (typically zeros) around the input to maintain its spatial dimensions after convolution. Padding can be used to prevent shrinking of the feature maps.
+
+- **Receptive Field**:  
+  The region of the input image that influences a particular feature in the output. Deeper layers in a CNN have a larger receptive field and can detect more complex features.
+
+- **Pooling**:  
+  Pooling layers downsample the feature maps by summarizing regions of the data. Max pooling selects the maximum value, while average pooling computes the average.
+
+### Steps Involved:
+1. **Input Data**:  
+   The input is typically an image or a grid-like structure. For example, an RGB image with dimensions 32x32x3.
+
+2. **Convolution**:  
+   The input is passed through convolutional layers, where multiple filters detect features like edges, corners, or textures.
+
+3. **Activation**:  
+   A non-linear activation function (ReLU) is applied to the convolved output, introducing non-linearity.
+
+4. **Pooling**:  
+   Pooling layers reduce the spatial dimensions of the feature maps, making the network computationally efficient.
+
+5. **Fully Connected Layers**:  
+   After several convolutional and pooling layers, the feature maps are flattened and passed through fully connected layers, producing a prediction.
+
+6. **Output**:  
+   For classification, the output is passed through a softmax layer, which converts the output into class probabilities.
+
+### Split Criteria:
+CNNs split data by progressively extracting features at different layers, with lower layers detecting simple patterns (e.g., edges) and higher layers capturing complex features (e.g., objects).
+
+### Time Complexity:
+- **Training Complexity**:  
+  The time complexity depends on the number of filters, kernel size, and input dimensions. For an image of size \( n \times n \) with \( k \) filters of size \( f \times f \), the time complexity of a convolutional layer is \( O(n^2 \cdot f^2 \cdot k) \).
```

Comment:
replace `\( n \times n \)` with `$ n \times n $`, `\( k \)` with `$k$`, `\( f \times f \)` with `$ f \times f $, and `\( O(n^2 \cdot f^2 \cdot k) \)` with `$ O(n^2 \cdot f^2 \cdot k)$`